syntax = 'proto3';

service Taskapi {
    rpc addTask (TaskDesc) returns (Id);
    rpc delTask(Id) returns (Task);
    rpc listTasks (Empty) returns (Tasks);
}

message TaskDesc {
    string description = 1;
}

message Task {
    int64 id = 1;
    string description = 2;
}

message Id {
    int64 id = 1;
}

message Tasks {
    repeated Task tasks = 1;
}

message Empty {
}
